3604/0/triangle.rb

```diff
@@ -0,0 +1,38 @@
+# Triangle Project Code.
+
+# Triangle analyzes the lengths of the sides of a triangle
+# (represented by a, b and c) and returns the type of triangle.
+#
+# It returns:
+#   :equilateral  if all sides are equal
+#   :isosceles    if exactly 2 sides are equal
+#   :scalene      if no sides are equal
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+# :reek:FeatureEnvy
+
+def test_error(val_a, val_b, val_c)
+  raise TriangleError if val_a <= 0 || val_b <= 0 || val_c <= 0
+  sum = (val_a + val_b + val_c) / 2.0
+  raise TriangleError if ((sum - val_a) * (sum - val_b) * (sum - val_c)) <= 0
+end
+# :reek:FeatureEnvy
+
+def triangle(val_a, val_b, val_c)
```

Comment:
What do you think about using OOP here? I do not want to stop you from thinking creatively, but I will share how it looks in my head. Your code shouldn't be the same, I will just try to put you on the path. 🙂
```
class Triangle
  def initialize(val_a, val_b, val_c)
    @val_a = val_a
    @val_b = val_b
    @val_c = val_c
  end

  def type
    # call method to check if it has errors or not
    # return type
  end
end

def triangle(val_a, val_b, val_c)
  Triangle.new(val_a, val_b, val_c).type
end
```

Reply:
Thanks, I see, I rewrite. :)